### ISSUE TITLE
Constructor property initialization

### ISSUE THREAD

**Issue by olivernybroe** [winglang/wing#5150]
### Feature Spec

Wing also has a concise syntax for declaring a constructor with properties and initializing them (works for both preflight and inflight):
```wing
class Bird {
    new(let type: str);
}
```

This can be combined with a body also
```wing
class Bird {
    name: str;

    new(let type: str, name: str) {
        this.name = name.uppercase();
    }
}
```

This concise syntax is a shorthand for the following:
```wing
class Bird {
   type: str;

    new(type: str) {
        this.type = type;
    }
}
```

### Use Cases

When having properties that just have a simple initialization, it will clean up the code and make it much more concise. 

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.
* If this issue is labeled **needs-discussion**, it means the spec has not been finalized yet. Please reach out on the #dev channel in the [Wing Slack](https://t.winglang.io/slack).

**Comment by yoav-steinberg** [comment]
Can you add an example of how to call a ctor with a prop initializer?
Is it like this:
```wing
new Bird("sparrow"); // Is this right?
new Bird(type="sparrow"); // Is this right?
```

**Comment by eladb** [comment]
@yoav-steinberg this is just a shortcut for defining fields.

It brings me to think that perhaps we should change the field definitions to use `let` as well.
But this is out of scope for this issue, so I opened a [discussion](https://github.com/winglang/wing/discussions/5162).

**Comment by Chriscbr** [comment]
```
class Bird {
    name: str;
 
    new(let type: str, name: str) {
        this.name = name.uppercase();
    }
}
```

In this example, would `type` be a private field by default? Is there a way for the user to specify that they want the field created for "type" to be `pub` or `protected` etc.?

**Comment by olivernybroe** [comment]
> ```
> class Bird {
>     name: str;
>  
>     new(let type: str, name: str) {
>         this.name = name.uppercase();
>     }
> }
> ```
> 
> In this example, would `type` be a private field by default? Is there a way for the user to specify that they want the field created for "type" to be `pub` or `protected` etc.?

Yep, following Wing convention with private by default this would be a private field.
You can add visibility, reassignment and variadic modifiers to it.
